database/src/migrations/20230817000000_critter_observation_deployment_tables.ts

```diff
@@ -0,0 +1,184 @@
+  create trigger audit_observation before insert or update or delete on observation for each row execute procedure tr_audit_trigger();
+  create trigger journal_observation after insert or update or delete on observation for each row execute procedure tr_journal_trigger();
+
+  ----------------------------------------------------------------------------------------
+  -- Create Critter table
+  ----------------------------------------------------------------------------------------
+
+  CREATE TABLE critter(
+    critter_id               integer           GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
+    observation_id           integer           NOT NULL,
+    critterbase_critter_id   uuid              NOT NULL,
+    create_date              timestamptz(6)    DEFAULT now() NOT NULL,
+    create_user              integer           NOT NULL,
+    update_date              timestamptz(6),
+    update_user              integer,
+    revision_count           integer           DEFAULT 0 NOT NULL,
+    CONSTRAINT critter_pk PRIMARY KEY (critter_id)
+  );
+
+  COMMENT ON COLUMN critter.critter_id               IS 'System generated surrogate primary key identifier.';
+  COMMENT ON COLUMN critter.observation_id           IS 'The id of the observation';
+  COMMENT ON COLUMN critter.critterbase_critter_id   IS 'The external system id of a Critterbase critter.';
+  COMMENT ON COLUMN critter.create_date              IS 'The datetime the record was created.';
+  COMMENT ON COLUMN critter.create_user              IS 'The id of the user who created the record as identified in the system user table.';
+  COMMENT ON COLUMN critter.update_date              IS 'The datetime the record was updated.';
+  COMMENT ON COLUMN critter.update_user              IS 'The id of the user who updated the record as identified in the system user table.';
+  COMMENT ON COLUMN critter.revision_count           IS 'Revision count used for concurrency control.';
+  COMMENT ON TABLE  critter                          IS 'Information about individual animals associated to specific observations.';
+
+  -- Add foreign key constraint from child table to parent table on observation_id
+  ALTER TABLE critter ADD CONSTRAINT critter_fk1
+    FOREIGN KEY (observation_id)
+    REFERENCES observation(observation_id);
+
+  -- Add unique constraint
+  CREATE UNIQUE INDEX critter_uk1 ON critter(critterbase_critter_id);
+
+  -- Create audit and journal triggers
+  create trigger audit_critter before insert or update or delete on critter for each row execute procedure tr_audit_trigger();
+  create trigger journal_critter after insert or update or delete on critter for each row execute procedure tr_journal_trigger();
+
+  ----------------------------------------------------------------------------------------
+  -- Create Deployment table
+  ----------------------------------------------------------------------------------------
+
+  CREATE TABLE deployment(
```

Comment:
Does this table need to track devices over time?

Not clear in my head if that is something sims should track, or if that is BCTW's responsibility which it already does.
Probably also relates back to that talk around whether or not its important to track which survey added a collar vs removed it, etc. So maybe for now we just don't worry about it until that is decided on.

Reply:
Yea im not too sure. I think just currently it needs to track the critter and the deployment. We will let BCTW manage all the collar information and deployment dates etc.

Still looking into the solution for tracking the survey for deployments and captures.